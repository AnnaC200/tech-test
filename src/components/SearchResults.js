import React from "react";
import PropTypes from "prop-types";
import "../styles/searchResults.css";

const SearchResults = ({ searchResults }) => {
  console.log(searchResults, "<-- results in search results")
  if (!searchResults.length) {
    return <p className="no-results">No results</p>
  } else {
    return (
      <>
        {searchResults.map((image) => {
          console.log(image, "<-- image")
          return <img className="search-results" src={image} alt="spaceImage" />
        })}
      </>
    );
  }
}

export default SearchResults;

SearchResults.propTypes = {
  searchResults: PropTypes.array.isRequired
}
