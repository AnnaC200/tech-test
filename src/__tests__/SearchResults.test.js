import React, { render } from '@testing-library/react';
import SearchResults from "../components/SearchResults";

describe("Search", () => {
    it("renders search result components correctly", () => {
        const { asFragment } = render(<SearchResults />);
        expect(asFragment()).toMatchSnapshot();
    });
});