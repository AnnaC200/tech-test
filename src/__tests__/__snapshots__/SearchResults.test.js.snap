// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search renders search result components correctly 1`] = `
<DocumentFragment>
  <img
    alt="spaceImage"
    class="search-results"
    src="moon"
  />
  <img
    alt="spaceImage"
    class="search-results"
    src="venus"
  />
</DocumentFragment>
`;
